daemons/controld/controld_messages.c

```diff
@@ -909,8 +909,12 @@ handle_node_info_request(const xmlNode *msg)
                            pcmk_is_set(controld_globals.flags,
                                        controld_has_quorum));
 
-    // Check whether client requested node info by ID and/or name
-    crm_element_value_int(msg, PCMK_XA_ID, &node_id);
+    /* Check whether client requested node info by ID and/or name
+     *
+     * @TODO A Corosync-layer node ID is of type uint32_t. We should be able to
+     * handle legitimate node IDs greater than INT_MAX, but currently we do not.
+     */
+    crm_element_value_int(msg, PCMK_XA_NODEID, &node_id);
```

Comment:
We can't change the attribute name because controller IPC is proxied to remote nodes (client and server may have different Pacemaker versions)

Reply:
boo